libs/remix-lib/src/web3Provider/web3VmProvider.ts

```diff
@@ -206,6 +210,29 @@ export class Web3VmProvider {
       error: data.error === false ? undefined : data.error
     }
     this.vmTraces[this.processingHash].structLogs.push(step)
+    // Track hardhat console.log call
+    if (step.op === 'STATICCALL' && step.stack[step.stack.length - 2] === '0x000000000000000000000000000000000000000000636f6e736f6c652e6c6f67') {
+      const stackLength = step.stack.length
+      const payloadStart = parseInt(step.stack[stackLength - 3], 16)
+      const memory = step.memory.join('')
+      let payload = memory.substring(payloadStart * 2, memory.length)
+      const fnselectorStr = payload.substring(0, 8)
+      const fnselectorStrInHex = '0x' + fnselectorStr
+      const fnselector = parseInt(fnselectorStrInHex)
+      const fnArgs = ConsoleLogs[fnselector]
+      const iface = new ethers.utils.Interface([`function log${fnArgs} view`])
+      const functionDesc = iface.getFunction(`log${fnArgs}`)
+      const sigHash = iface.getSighash(`log${fnArgs}`)
+      if (fnArgs.includes('uint') && sigHash !== fnselectorStrInHex) {
+        payload = payload.replace(fnselectorStr, sigHash)
```

Comment:
you don't need to prepend `0x` in that case?

Reply:
No, sigHash returned from `iface.getSighash(`log${fnArgs}`)` already contains it